browser/js/modalWindow/modal.controller.js

```diff
@@ -0,0 +1,28 @@
+app.controller('ModalInstanceCtrl', function ($scope, $uibModalInstance, ModalFactory) {
+
+  // $scope.items = items;
+  // $scope.selected = {
+  //   item: $scope.items[0]
+  // };
+
+  $scope.ok = function () {
+    $uibModalInstance.close();
+  };
+
+  $scope.cancel = function () {
+    $uibModalInstance.dismiss('cancel');
+  };
+
+  $scope.post= function(){
+    $scope.kit.categories= $scope.kit.categories.split(",");
+    ModalFactory.postKit($scope.kit)
+    .then(()=> {
+     $scope.ok();
+     // $scope.$digest();  how do i make the digest work upon adding a new page?
+    })
+    .catch(function (err) {
+      console.log('SOMETHING WENT TERRIBLY WRONG');
```

Comment:
Not necessary to change, but I think you can use .catch($log.error); here to shorten your code. You'll need to inject $log into the controller though.